pkgs/native_assets_cli/lib/src/model/build_output.dart

```diff
@@ -2,143 +2,176 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'dart:io';
-
-import 'package:collection/collection.dart';
-import 'package:pub_semver/pub_semver.dart';
-import 'package:yaml/yaml.dart';
-
-import '../api/build_output.dart' as api;
-import '../utils/datetime.dart';
-import '../utils/file.dart';
-import '../utils/map.dart';
-import '../utils/yaml.dart';
-import 'asset.dart';
-import 'dependencies.dart';
-import 'metadata.dart';
-
-class BuildOutput implements api.BuildOutput {
-  /// Time the build this output belongs to started.
-  ///
-  /// Rounded down to whole seconds, because [File.lastModified] is rounded
-  /// to whole seconds and caching logic compares these timestamps.
+part of '../api/build_output.dart';
+
+final class BuildOutputImpl implements BuildOutput {
   @override
   final DateTime timestamp;
+
+  final List<AssetImpl> _assets;
+
   @override
-  final List<Asset> assets;
-  @override
-  final Dependencies dependencies;
+  Iterable<AssetImpl> get assets => _assets;
+
+  final Dependencies _dependencies;
+
+  Dependencies get dependenciesModel => _dependencies;
```

Comment:
Does this need to be public?

Reply:
But the *object* is exposed, so someone can cast to `dynamic` and read `dependenciesModel` from the object.
I strongly recommend not exposing public instance members on implementation classes, and relying on the *type* being private for hiding them. Just make them private. If you need to access them, have a helper function to access them.

Fx here, make it:
```dart
static Dependencies dependenciesModelOf(BuildOutputImpl output) => output._dependencies;
```

Then someone has to actually import the `BuildOutputImpl` class, inside `src/`, to get to the value, not just cast to `dynamic`.